src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1934,8 +1934,7 @@ public StringBuilder Replace(string oldValue, string? newValue, int startIndex,
 
             newValue ??= string.Empty;
 
-            Span<int> replacements = stackalloc int[5]; // A list of replacement positions in a chunk to apply
-            int replacementsCount = 0;
+            var replacements = new ValueListBuilder<int>(stackalloc int[128]); // A list of replacement positions in a chunk to apply
```

Comment:
great use case for `ValueListBuilder` 👍 I am suprised that the temporary array allocation survived for so long in this repo ;)